### PR TITLE
template: add tests

### PR DESCRIPTION
**Pull request by simonpasquier**
One small improvement and more tests to make sure that it is valid.

### CODE REVIEW

File: template/template.go [hunk 1]
```diff
@@ -15,16 +15,15 @@ package template
 
 import (
 	"bytes"
+	tmplhtml "html/template"
```
Comment (stuartnelson3) [inline template/template.go]: this was formatted like this by `goimports`? weird

Comment (simonpasquier) [reply template/template.go]: It was most probably a manual renaming.

File: template/template.go [hunk 2]
```diff
@@ -325,6 +324,9 @@ func (t *Template) Data(recv string, groupLabels model.LabelSet, alerts ...*type
 					delete(commonAnnotations, an)
 				}
 			}
+			if len(commonLabels) == 0 && len(commonAnnotations) == 0 {
```
Comment (mxinden) [inline template/template.go]: Why not move this below `for _, a := range alerts[1:] {`? Am I missing something?

Comment (simonpasquier) [reply template/template.go]: Yes it works too.

**Pull request merged**